fw/vhost.c

```diff
@@ -2729,60 +2965,77 @@ static TfwCfgSpec tfw_global_frang_specs[] = {
 		},
 		.allow_reconfig = true,
 	},
-	/* Option can be redefined per vhost|location. */
+	/* Option can be redefined per vhost|location.
+	 *
+	 *  All handler are changed to tfw_cfgop_frang_glob_...
+	 *  because here we need to compare the default value
+	 *  with 0 after kzalloc, and for non global configuration
+	 *  we need to already check default value with real default
+	 *  value set here or in tfw_vhost_frang_specs
+	 */
 	{
 		.name = "http_uri_len",
 		.deflt = "0",
-		.handler = tfw_cfgop_frang_uri_len,
+		.handler = tfw_cfgop_frang_glob_set_int,
+		.dest = &tfw_frang_vhost_reconfig.http_uri_len,
 		.allow_reconfig = true,
 	},
 	{
 		.name = "http_body_len",
 		.deflt = "1073741824", /* 1 Gb. */
-		.handler = tfw_cfgop_frang_body_len,
+		.handler = tfw_cfgop_frang_glob_set_long,
```

Comment:
We have inconsistency behavior here.
```
frang_limits {
    http_body_len 0;
}
 frang_limits {
 }
 ```
I have `http_body_len=1073741824` (I think that we should 0), because default value should not override value which was set previously. For
```
frang_limits {
    http_ct_required true;
}
 frang_limits {
}
```
I have  http_ct_required=true. 

I think we have the same problem for `concurrent_tcp_connections`, `http_header_cnt`, `http_body_len`, `http_strict_host_checking`, `http_methods`. For example in vhost section you check  if (ce->dflt_value && cfg->http_body_len != deflt) for `http_body_len` but in global section if (ce->dflt_value && *(unsigned int *)(cs->dest). But ce->dest can be 0, which is specified by administrator!